velox/common/memory/MemoryPool.h

```diff
@@ -277,10 +336,11 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
   const Kind kind_;
   const uint16_t alignment_;
   const std::shared_ptr<MemoryPool> parent_;
+  const std::shared_ptr<MemoryReclaimer> reclaimer_;
 
   /// Protects 'children_'.
   mutable folly::SharedMutex childrenMutex_;
-  std::unordered_map<std::string, std::weak_ptr<MemoryPool>> children_;
+  std::unordered_map<std::string, MemoryPool*> children_;
```

Comment:
nit: perhaps, add a comment to explain why you decided not to use weak_ptr. This would help future readers who may wonder why we don't do that.

Reply:
Added. Thanks!